### DISCUSSION TITLE
API to make a copy of an existing dashboard

### DISCUSSION THREAD

**Discussion by axoxymous-coder** [apache/superset#24577] [Q&A / Help]
As per the Superset API documentation, `POST /api/v1/dashboard/{id_or_slug}/copy/` can be used to make a copy of an existing dashboard (Save As of a dashboard from the UI)

The API takes the below payload:
```
{
  "css": "string",
  "dashboard_title": "string",
  "duplicate_slices": true,
  "json_metadata": "string"
}
```

Where can I get the necessary `json_metadata` as required by this API? I found that a part of this can be found in the response of `GET /api/v1/dashboard/{id_or_slug}` API, specifically the `json_metadata` and `position_json`. However, the copy dashboard API also required some data w.r.t filters and filter scopes. I am not able to find where can I find those data so as to pass a full fledged payload to the copy API.

Any help will be highly appreciated.

**Comment by Alexey-Khromenko** [comment]
@axoxymous-coder, hello
I found that it's possible to use folowing payload:
```
{
  "css": "",
  "duplicate_slices": False,
  "dashboard_title": "test",
  "json_metadata": "{}"
}
```

new dashboard appears without formating. Method "update" work fine to add  "position_json" data (formating).

If   we use **"duplicate_slices": True** - server return error 500 "Fatal error" (though id counter for slices (charts) growths every time)

Does anyone face this issue?

**Reply by Parzi68** [reply]
yess.... Im also facing the same issue...

**Comment by BrenoOsvaldoFunicheli** [comment]
I had this problem too. I can't found any solution, until now

**Comment by Parzi68** [comment]
I think it is because of the position of the charts.

**Comment by shahargl** [comment]
I've managed to solve it:
```python
 # Get the dashboard details to extract json_metadata
dashboard_details_response = requests.get(
    f"{self.base_url}/api/v1/dashboard/{dashboard_id}",
    headers=self.get_headers(),
    cookies=self.get_cookies(),
)
dashboard_details_response.raise_for_status()
# get the metadata and positions
dashboard_metadata_json = dashboard_details_response.json().get("result", {}).get("json_metadata")
dashboard_positions = dashboard_details_response.json().get("result", {}).get("position_json")

# Update the metadata with positions - it needs to be a dict
metadata_dict = json.loads(dashboard_metadata_json)
metadata_dict["positions"] = json.loads(dashboard_positions)

# Copy the dashboard using the API
copy_payload = {
    "dashboard_title": new_dashboard_title,
    "duplicate_slices": True,
    "json_metadata": json.dumps(metadata_dict)
}

response = requests.post(
    f"{self.base_url}/api/v1/dashboard/{dashboard_id}/copy/",
    headers=self.get_headers(),
    cookies=self.get_cookies(),
    json=copy_payload
)
```
